api/db-migrations/000009_ensembling_job.up.sql

```diff
@@ -18,6 +18,8 @@ CREATE TABLE IF NOT EXISTS ensembling_jobs
     environment_name varchar(50) NOT NULL,
     infra_config     jsonb NOT NULL,
     job_config       jsonb NOT NULL,
+    is_locked        bool NOT NULL default 'false',
```

Comment:
When are multiple instances of Turing deployed at once, the batch job runner is able to pick up the same job for processing. We want to set a lock status so that only one record is picked up at once (https://github.com/ashwinath/turing/blob/batch-ensembler-orchestration/api/turing/batch/runner/ensembling.go#L206).

Reply:
This flag is not exposed to the user.

Reply:
In what (legit) scenarios do you expect to have multiple instances of Turing deployments? 
Can you add a comment, that describes that `is_locked` is set to True for the time it's being processed by the job runner?

Reply:
During rolling updates thats for one (there will be a point where theres more than 1), unless we are doing `Recreate` rollout strategy. Not too hard to support it anyway, the other way is to use a mutex as there will be some kind of locking mechanism required.

Anyway the other scenario is explained here even during single deployments: https://github.com/gojek/turing/pull/61#discussion_r656779512

Reply:
yes, I can add in a comment describing this. Code wise the comment is already there but I will add it in the SQL.

Reply:
I think a mutex would be ideal to coordinate this DB lock. CMIIW: https://github.com/gojek/turing/pull/61#discussion_r658412168

Reply:
I've been thinking a little bit about this. Do we need the explicit `isLocked` in the DB? Would it not be enough to look at `status`? We are picking up jobs in pending / one of the problematic terminal states [here](https://github.com/ashwinath/turing/blob/batch-ensembler-orchestration/api/turing/batch/runner/ensembling.go#L207) and updating the field to 'building' (along with the `isLocked`) [here](https://github.com/ashwinath/turing/blob/batch-ensembler-orchestration/api/turing/batch/runner/ensembling.go#L244). Before the job is unlocked, we are once again updating its status. Am I missing something?

Reply:
Actually you are right, I think when I first designed this, I was thinking along the lines of `update ensembling_jobs set is_locked = true where status = 'pending'  and is_locked = false returning *`. But my plans were kinda messed up by the gorm situation. I think its ok to remove the is_locked if thats the case.

Reply:
I have removed the is_locked flag completely. Thanks for pointing this out.